pkg/volume/empty_dir/empty_dir.go

```diff
@@ -133,6 +136,7 @@ type emptyDir struct {
 	mountDetector mountDetector
 	plugin        *emptyDirPlugin
 	rootContext   string
+	volume.MetricsDu
```

Comment:
minor idea: rather than embedding MetricsDu and then calling Init(), why not embed `MetricsProvider` and initialize it in the struct definition as `MetricsProvider: volume.NewMetricsDu(path)` ?